src/modules/TodoListArray.js

```diff
@@ -0,0 +1,53 @@
+class TodoListArray {
+  constructor() {
+    this.todos = JSON.parse(localStorage.getItem('todos')) || [];
+  }
+
+  addTodo(todo) {
+    this.todos.push(todo);
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  updateTodo(idx, newTodo) {
+    this.todos[idx - 1] = newTodo;
+    for (let i = 0; i < this.todos.length; i += 1) {
+      this.todos[i].idx = i + 1;
+    }
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  deleteTodo(index) {
+    if (this.todos.length === 1) {
+      this.todos = [];
+    } else {
+      this.todos.splice(index, 1);
+    }
+    for (let i = 0; i < this.todos.length; i += 1) {
+      this.todos[i].idx = i + 1;
+    }
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  clickCompleted(index) {
+    this.todos[index - 1].completed = !this.todos[index - 1].completed;
```

Comment:
- [ ] kindly try to use the Strict equality operator (“!==”) instead of the Loose Equality operator (‘!=’), to ensure you get the right result.

[for more information please follow this link](https://medium.com/swlh/equality-operator-in-javascript-2d4055c8464d#:~:text=Implicit%20type%20coercion%20is%20when,converted%20between%20different%20types%20automatically.&text=Explicit%20type%20coercion%20is%20when,from%20one%20type%20to%20another.&text=Equality%20operators%20are%20part%20of,to%20either%20true%20or%20false.)

Reply:
Thanks for pointing this out, but I'm not checking for equality rather I assigned it to the inverse value.

Reply:
Hey @UsirusenIkon 👋, 
Thank you for that insight😎👏👏 , please ignore the comment above.

Happy coding 👨‍💻.